File: C1.UWP.InputPanel/CS/InputPanelSamples/Common/CommandHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Input;

namespace InputPanelSamples
{
    public class DelegateCommand : ICommand
    {
        private Action _action;
        private bool _canExecute = false;

        public DelegateCommand(Action action, bool canExecute)
        {
            this._action = action;
            this._canExecute = canExecute;
        }

        public bool CanExecute(object parameter)
        {
            return _canExecute;
        }

        public void Execute(object parameter)
        {
            _action();
        }

        public event EventHandler CanExecuteChanged;

        public void OnCanExecuteChanged()
        {
            CanExecuteChanged(this, EventArgs.Empty);
        }
    }
}

File: C1.UWP.InputPanel/VB/InputPanelSamples/Common/CommandHelper.vb
Namespace Common
	Public Class DelegateCommand
		Implements ICommand
		Private _action As Action
		Private _canExecute As Boolean = False

        Public Sub New(action As Action, canExecute As Boolean)
            Me._action = action
            Me._canExecute = canExecute
        End Sub

        Private Event ICommand_CanExecuteChanged As EventHandler Implements ICommand.CanExecuteChanged

        Private Function ICommand_CanExecute(parameter As Object) As Boolean Implements ICommand.CanExecute
            Return _canExecute
        End Function

        Private Sub ICommand_Execute(parameter As Object) Implements ICommand.Execute
            _action()
        End Sub
    End Class
End Namespace